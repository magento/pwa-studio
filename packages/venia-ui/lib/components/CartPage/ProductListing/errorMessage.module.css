.root {
    border-left: 5px solid rgb(var(--venia-global-color-error));
    color: rgb(var(--venia-global-color-error));
    font-size: var(--venia-typography-body-S-fontSize);
    font-weight: var(--venia-global-fontWeight-semibold);
    line-height: var(--venia-global-lineHeight-300);
    padding: 1rem 0 1rem 1rem;
    margin-bottom: 1rem;
}
