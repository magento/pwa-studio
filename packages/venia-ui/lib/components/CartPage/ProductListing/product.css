.root {
    display: grid;
    row-gap: 0.5rem;
}

.item {
    align-items: start;
    display: grid;
    grid-gap: 0.5rem 1rem;
    grid-template-areas: 'image details kebab';
    grid-template-columns: 100px 1fr min-content;
}

.item_disabled {
    composes: item;
    opacity: 0.5;
}

.errorText {
    color: rgb(var(--venia-global-color-error));
    line-height: var(--venia-global-lineHeight-300);
}

.errorText:empty {
    display: none;
}

.imageContainer {
    grid-area: image;
}

.imageRoot {
    height: 100%;
}

.image {
    background-color: rgb(var(--venia-global-color-gray));
    border: solid 1px rgb(var(--venia-global-color-border));
    border-radius: 2px;
    height: 100%;
    object-fit: contain;
    object-position: center;
}

.details {
    display: grid;
    gap: 0.5rem;
    grid-area: details;
    grid-template-areas:
        'name name'
        'options quantity'
        'price quantity'
        'stock quantity';
    grid-template-columns: 2fr 1fr;
    line-height: 1.5;
}

.name {
    grid-area: name;
    font-weight: var(--venia-global-fontWeight-semibold);
}

.price {
    grid-area: price;
    font-size: var(--venia-typography-body-S-fontSize);
}

.quantity {
    align-items: start;
    display: grid;
    grid-area: quantity;
    justify-items: center;
}

.kebab {
    grid-area: kebab;
    position: relative;
}

.sectionText {
    padding: 0.25rem 0.5rem;
    font-size: var(--venia-typography-body-S-fontSize);
    pointer-events: none;
}

.options {
    grid-area: options;
    font-size: var(--venia-typography-body-S-fontSize);
    display: grid;
    grid-gap: 0.125rem;
}

.optionLabel {
    display: grid;
    grid-auto-flow: column;
    grid-auto-columns: max-content;
}

.stockStatusMessage {
    color: rgb(var(--venia-global-color-error));
    font-size: var(--venia-global-fontSize-100);
    font-weight: var(--venia-global-fontWeight-semibold);
    grid-area: stock;
}

.stockStatusMessage:empty {
    display: none;
}

.addToListButton {
    --stroke: rgb(var(--venia-global-color-teal));
    align-content: center;
    column-gap: 1.125rem;
    display: inline-flex;
    font-size: var(--venia-typography-body-S-fontSize);
    padding: 0.875rem 0.625rem;
    width: 100%;
}

.addToListButton_selected {
    display: none;
}

@media (max-width: 640px) {
    .details {
        grid-template: none;
    }

    .name,
    .options,
    .price,
    .stockStatusMessage,
    .quantity {
        grid-area: auto;
    }

    .quantity {
        justify-items: start;
    }
}
