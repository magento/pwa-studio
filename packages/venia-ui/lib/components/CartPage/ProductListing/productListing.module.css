.root {
    display: grid;
    grid-gap: 2rem 1rem;
}

.errorText {
    color: rgb(var(--venia-global-color-error));
    line-height: var(--venia-global-lineHeight-300);
    margin-bottom: 1rem;
}
