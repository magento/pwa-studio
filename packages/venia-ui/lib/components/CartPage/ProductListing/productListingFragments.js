import { gql } from '@apollo/client';

export const ProductListingFragment = gql`
    fragment ProductListingFragment on Cart {
        id
        items {
            id
            product {
                id
                name
                sku
                url_key
                url_suffix
                thumbnail {
                    url
                }
                small_image {
                    url
                }
                stock_status
                ... on ConfigurableProduct {
                    variants {
                        attributes {
                            uid
                        }
                        product {
                            id
                            small_image {
                                url
                            }
                        }
                    }
                }
            }
            prices {
                price {
                    currency
                    value
                }
            }
            quantity
            ... on ConfigurableCartItem {
                configurable_options {
                    id
                    configurable_product_option_value_uid
                    option_label
                    value_id
                    value_label
                }
            }
        }
    }
`;
