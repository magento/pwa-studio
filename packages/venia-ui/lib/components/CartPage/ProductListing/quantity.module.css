.root {
    --stroke: var(--venia-global-color-gray-600);
    text-align: center;
    display: grid;
    grid-template-columns: 1fr 4rem 1fr;
    grid-column-gap: 0.5rem;
    align-items: center;
    justify-items: center;
}

.wrap {
    display: flex;
    align-items: center;
    justify-content: center;
}

.label:not(:focus):not(:active) {
    position: absolute;
    overflow: hidden;
    width: 1px;
    height: 1px;
    clip: rect(0 0 0 0);
    clip-path: inset(100%);
    white-space: nowrap;
}

.button {
    align-items: center;
    background-color: rgb(var(--venia-global-color-gray-50));
    border: 2px solid rgb(var(--venia-global-color-gray-600));
    border-radius: 50%;
    display: inline-flex;
    height: 2rem;
    justify-content: center;
    stroke: rgb(var(--stroke));
    width: 2rem;
}

.button:hover {
    --stroke: rgb(var(--venia-global-color-gray-900));
    border-color: rgb(var(--stroke));
}

.button:disabled,
.button:disabled:hover {
    --stroke: rgb(var(--venia-global-color-gray-400));
    border-color: rgb(var(--venia-global-color-gray-400));
}

.button:disabled {
    cursor: not-allowed;
}

.button:focus {
    outline: none;
}

.button_decrement {
    composes: button;
}

.button_increment {
    composes: button;
}

.icon {
    align-items: center;
    display: inline-flex;
    justify-content: center;
    touch-action: manipulation;
}

.input {
    composes: input from '../../TextInput/textInput.module.css';
    text-align: center;
}
