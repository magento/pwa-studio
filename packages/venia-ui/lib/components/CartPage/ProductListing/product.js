import React, { useMemo } from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { Heart } from 'react-feather';
import { gql } from '@apollo/client';
import { Link } from 'react-router-dom';
import { useProduct } from '@magento/peregrine/lib/talons/CartPage/ProductListing/useProduct';
import resourceUrl from '@magento/peregrine/lib/util/makeUrl';
import Price from '@magento/venia-ui/lib/components/Price';

import { useStyle } from '../../../classify';
import Icon from '../../Icon';
import Image from '../../Image';
import Kebab from '../../LegacyMiniCart/kebab';
import ProductOptions from '../../LegacyMiniCart/productOptions';
import Section from '../../LegacyMiniCart/section';
import AddToListButton from '../../Wishlist/AddToListButton';
import Quantity from './quantity';

import defaultClasses from './product.css';

import { CartPageFragment } from '../cartPageFragments.gql';
import { AvailableShippingMethodsCartFragment } from '../PriceAdjustments/ShippingMethods/shippingMethodsFragments.gql';

const IMAGE_SIZE = 100;

const HeartIcon = <Icon size={16} src={Heart} />;

const Product = props => {
    const { item } = props;

    const { formatMessage } = useIntl();
    const talonProps = useProduct({
        operations: {
            removeItemMutation: REMOVE_ITEM_MUTATION,
            updateItemQuantityMutation: UPDATE_QUANTITY_MUTATION
        },
        ...props
    });

    const {
        addToWishlistProps,
        errorMessage,
        handleEditItem,
        handleRemoveFromCart,
        handleUpdateItemQuantity,
        isEditable,
        product,
        isProductUpdating
    } = talonProps;

    const {
        currency,
        image,
        name,
        options,
        quantity,
        stockStatus,
        unitPrice,
        urlKey,
        urlSuffix
    } = product;

    const classes = useStyle(defaultClasses, props.classes);

    const itemClassName = isProductUpdating
        ? classes.item_disabled
        : classes.item;

    const editItemSection = isEditable ? (
        <Section
            text={formatMessage({
                id: 'product.editItem',
                defaultMessage: 'Edit item'
            })}
            onClick={handleEditItem}
            icon="Edit2"
            classes={{
                text: classes.sectionText
            }}
        />
    ) : null;

    const itemLink = useMemo(
        () => resourceUrl(`/${urlKey}${urlSuffix || ''}`),
        [urlKey, urlSuffix]
    );

    const stockStatusMessage =
        stockStatus === 'OUT_OF_STOCK'
            ? formatMessage({
                  id: 'product.outOfStock',
                  defaultMessage: 'Out-of-stock'
              })
            : '';

    return (
        <li className={classes.root}>
            <span className={classes.errorText}>{errorMessage}</span>
            <div className={itemClassName}>
                <Link to={itemLink} className={classes.imageContainer}>
                    <Image
                        alt={name}
                        classes={{
                            root: classes.imageRoot,
                            image: classes.image
                        }}
                        width={IMAGE_SIZE}
                        resource={image}
                    />
                </Link>
                <div className={classes.details}>
                    <div className={classes.name}>
                        <Link to={itemLink}>{name}</Link>
                    </div>
                    <ProductOptions
                        options={options}
                        classes={{
                            options: classes.options,
                            optionLabel: classes.optionLabel
                        }}
                    />
                    <span className={classes.price}>
                        <Price currencyCode={currency} value={unitPrice} />
                        <FormattedMessage
                            id={'product.price'}
                            defaultMessage={' ea.'}
                        />
                    </span>
                    <span className={classes.stockStatusMessage}>
                        {stockStatusMessage}
                    </span>
                    <div className={classes.quantity}>
                        <Quantity
                            itemId={item.id}
                            initialValue={quantity}
                            onChange={handleUpdateItemQuantity}
                        />
                    </div>
                </div>
                <Kebab
                    classes={{
                        root: classes.kebab
                    }}
                    disabled={true}
                >
                    {editItemSection}
                    <Section
                        text={formatMessage({
                            id: 'product.removeFromCart',
                            defaultMessage: 'Remove from cart'
                        })}
                        onClick={handleRemoveFromCart}
                        icon="Trash"
                        classes={{
                            text: classes.sectionText
                        }}
                    />
                    <li>
                        <AddToListButton
                            {...addToWishlistProps}
                            classes={{
                                root: classes.addToListButton,
                                root_selected: classes.addToListButton_selected
                            }}
                            icon={HeartIcon}
                        />
                    </li>
                </Kebab>
            </div>
        </li>
    );
};

export default Product;

export const REMOVE_ITEM_MUTATION = gql`
    mutation removeItem($cartId: String!, $itemId: Int!) {
        removeItemFromCart(input: { cart_id: $cartId, cart_item_id: $itemId })
            @connection(key: "removeItemFromCart") {
            cart {
                id
                ...CartPageFragment
                ...AvailableShippingMethodsCartFragment
            }
        }
    }
    ${CartPageFragment}
    ${AvailableShippingMethodsCartFragment}
`;

export const UPDATE_QUANTITY_MUTATION = gql`
    mutation updateItemQuantity(
        $cartId: String!
        $itemId: Int!
        $quantity: Float!
    ) {
        updateCartItems(
            input: {
                cart_id: $cartId
                cart_items: [{ cart_item_id: $itemId, quantity: $quantity }]
            }
        ) @connection(key: "updateCartItems") {
            cart {
                id
                ...CartPageFragment
                ...AvailableShippingMethodsCartFragment
            }
        }
    }
    ${CartPageFragment}
    ${AvailableShippingMethodsCartFragment}
`;
