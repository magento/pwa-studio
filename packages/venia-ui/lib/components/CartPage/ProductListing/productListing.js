import React, { Fragment, Suspense } from 'react';
import { FormattedMessage } from 'react-intl';
import { gql } from '@apollo/client';
import { useProductListing } from '@magento/peregrine/lib/talons/CartPage/ProductListing/useProductListing';

import { useStyle } from '../../../classify';
import LoadingIndicator from '../../LoadingIndicator';
import defaultClasses from './productListing.css';
import Product from './product';
import { ProductListingFragment } from './productListingFragments';

const EditModal = React.lazy(() => import('./EditModal'));
/**
 * A child component of the CartPage component.
 * This component renders the product listing on the cart page.
 *
 * @param {Object} props
 * @param {Function} props.setIsCartUpdating Function for setting the updating state of the cart.
 * @param {Object} props.classes CSS className overrides.
 * See [productListing.css]{@link https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/ProductListing/productListing.css}
 * for a list of classes you can override.
 *
 * @returns {React.Element}
 *
 * @example <caption>Importing into your project</caption>
 * import ProductListing from "@magento/venia-ui/lib/components/CartPage/ProductListing";
 */
const ProductListing = props => {
    const {
        onAddToWishlistSuccess,
        setIsCartUpdating,
        fetchCartDetails
    } = props;
    const talonProps = useProductListing({
        queries: {
            getProductListing: GET_PRODUCT_LISTING
        }
    });
    const {
        activeEditItem,
        isLoading,
        items,
        setActiveEditItem,
        wishlistConfig
    } = talonProps;

    const classes = useStyle(defaultClasses, props.classes);

    if (isLoading) {
        return (
            <LoadingIndicator>
                <FormattedMessage
                    id={'productListing.loading'}
                    defaultMessage={'Fetching Cart...'}
                />
            </LoadingIndicator>
        );
    }

    if (items.length) {
        const productComponents = items.map(product => (
            <Product
                item={product}
                key={product.id}
                setActiveEditItem={setActiveEditItem}
                setIsCartUpdating={setIsCartUpdating}
                onAddToWishlistSuccess={onAddToWishlistSuccess}
                fetchCartDetails={fetchCartDetails}
                wishlistConfig={wishlistConfig}
            />
        ));

        return (
            <Fragment>
                <ul className={classes.root}>{productComponents}</ul>
                <Suspense fallback={null}>
                    <EditModal
                        item={activeEditItem}
                        setIsCartUpdating={setIsCartUpdating}
                        setActiveEditItem={setActiveEditItem}
                    />
                </Suspense>
            </Fragment>
        );
    }

    return null;
};

export const GET_PRODUCT_LISTING = gql`
    query getProductListing($cartId: String!) {
        cart(cart_id: $cartId) {
            id
            ...ProductListingFragment
        }
    }
    ${ProductListingFragment}
`;

export default ProductListing;
