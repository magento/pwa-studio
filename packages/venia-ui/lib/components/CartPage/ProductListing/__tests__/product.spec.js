import React from 'react';
import { createTestInstance } from '@magento/peregrine';

import Product from '../product';
import { useProduct } from '@magento/peregrine/lib/talons/CartPage/ProductListing/useProduct';

jest.mock('../../../Image', () => 'Image');
jest.mock('@magento/peregrine/lib/talons/CartPage/ProductListing/useProduct');
jest.mock('../../../../classify');
jest.mock('@apollo/client', () => {
    const executeMutation = jest.fn(() => ({ error: null }));
    const useMutation = jest.fn(() => [executeMutation]);

    return {
        gql: jest.fn(),
        useMutation
    };
});

jest.mock('../../../Wishlist/AddToListButton', () => 'AddToListButton');

jest.mock('@magento/peregrine/lib/context/cart', () => {
    const state = { cartId: 'cart123' };
    const api = {};
    const useCartContext = jest.fn(() => [state, api]);

    return { useCartContext };
});

jest.mock('react-router-dom', () => ({
    Link: ({ children, ...rest }) => <div {...rest}>{children}</div>
}));
jest.mock('@magento/peregrine/lib/util/makeUrl');

const props = {
    item: {
        id: '123',
        product: {
            name: 'Unit Test Product',
            small_image: {
                url: 'unittest.jpg'
            },
            urlKey: 'unittest',
            urlSuffix: '.html',
            sku: '12345'
        },
        prices: {
            price: {
                currency: 'USD',
                value: 100
            }
        },
        quantity: 1,
        configurable_options: [
            {
                configurable_product_option_value_uid: '12345asd'
            },
            {
                configurable_product_option_value_uid: 'asf2134'
            }
        ]
    }
};

test('renders simple product correctly', () => {
    useProduct.mockReturnValueOnce({
        addToWishlistProps: {
            atwProp1: 'value1'
        },
        errorMessage: undefined,
        handleEditItem: jest.fn(),
        handleRemoveFromCart: jest.fn(),
        handleUpdateItemQuantity: jest.fn(),
        isEditable: false,
        product: {
            currency: 'USD',
            image: {},
            name: '',
            options: [],
            quantity: 1,
            unitPrice: 1,
            urlKey: 'unittest',
            urlSuffix: '.html'
        },
        isProductUpdating: false
    });
    const tree = createTestInstance(<Product {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
});

test('renders out of stock product', () => {
    useProduct.mockReturnValueOnce({
        addToWishlistProps: {
            atwProp1: 'value1'
        },
        errorMessage: undefined,
        handleEditItem: jest.fn(),
        handleRemoveFromCart: jest.fn(),
        handleUpdateItemQuantity: jest.fn(),
        isEditable: false,
        product: {
            currency: 'USD',
            image: {},
            name: '',
            options: [],
            quantity: 2,
            stockStatus: 'OUT_OF_STOCK',
            unitPrice: 55,
            urlKey: 'popular-product',
            urlSuffix: ''
        },
        loginToastProps: null,
        isProductUpdating: false
    });
    const tree = createTestInstance(<Product {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
});

test('renders configurable product with options', () => {
    useProduct.mockReturnValueOnce({
        addToWishlistProps: {
            atwProp1: 'value1'
        },
        errorMessage: undefined,
        handleEditItem: jest.fn(),
        handleRemoveFromCart: jest.fn(),
        handleUpdateItemQuantity: jest.fn(),
        isEditable: true,
        product: {
            currency: 'USD',
            image: {},
            name: '',
            urlKey: 'unittest',
            urlSuffix: '.html',
            options: [
                {
                    option_label: 'Option 1',
                    value_label: 'Value 1'
                },
                {
                    option_label: 'Option 2',
                    value_label: 'Value 2'
                }
            ],
            quantity: 1,
            unitPrice: 1
        },
        loginToastProps: null,
        isProductUpdating: false
    });

    const tree = createTestInstance(<Product {...props} />);

    expect(tree.toJSON()).toMatchSnapshot();
});
