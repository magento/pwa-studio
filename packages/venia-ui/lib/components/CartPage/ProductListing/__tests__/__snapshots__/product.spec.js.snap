// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders configurable product with options 1`] = `
<li
  className="root"
>
  <span
    className="errorText"
  />
  <div
    className="item"
  >
    <div
      className="imageContainer"
      to="/unittest.html"
    >
      <Image
        alt=""
        classes={
          Object {
            "image": "image",
            "root": "imageRoot",
          }
        }
        resource={Object {}}
        width={100}
      />
    </div>
    <div
      className="details"
    >
      <div
        className="name"
      >
        <div
          to="/unittest.html"
        >
          
        </div>
      </div>
      <dl
        className="options"
      >
        <div
          className="optionLabel"
        >
          <dt
            className="optionName"
          >
            Option 1 :
          </dt>
          <dd
            className="optionValue"
          >
            Value 1
          </dd>
        </div>
        <div
          className="optionLabel"
        >
          <dt
            className="optionName"
          >
            Option 2 :
          </dt>
          <dd
            className="optionValue"
          >
            Value 2
          </dd>
        </div>
      </dl>
      <span
        className="price"
      >
        <span>
          $
        </span>
        <span>
          1
        </span>
        <span>
          .
        </span>
        <span>
          00
        </span>
        <mock-FormattedMessage
          defaultMessage=" ea."
          id="product.price"
        />
      </span>
      <span
        className="stockStatusMessage"
      >
        
      </span>
      <div
        className="quantity"
      >
        <form
          onKeyDown={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="root"
          >
            <label
              className="label"
              htmlFor="123"
            >
              Quantity
            </label>
            <button
              aria-label="Decrease Quantity"
              className="button_decrement"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              <span
                className="root"
              >
                <svg
                  className="icon"
                  fill="none"
                  height={22}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={22}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
            <span
              className="root"
              style={
                Object {
                  "--iconsAfter": 0,
                  "--iconsBefore": 0,
                }
              }
            >
              <span
                className="input"
              >
                <input
                  aria-label="Item Quantity"
                  className="input"
                  id="123"
                  inputMode="numeric"
                  min={0}
                  name="quantity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  pattern="[0-9]*"
                  value={1}
                />
              </span>
              <span
                className="before"
              />
              <span
                className="after"
              />
            </span>
            <p
              className="root"
            />
            <button
              aria-label="Increase Quantity"
              className="button_increment"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="root"
              >
                <svg
                  className="icon"
                  fill="none"
                  height={20}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="12"
                    x2="12"
                    y1="5"
                    y2="19"
                  />
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="root"
    >
      <button
        className="kebab"
        onClick={[Function]}
      >
        <span
          className="root"
        >
          <svg
            className="icon"
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="12"
              cy="5"
              r="1"
            />
            <circle
              cx="12"
              cy="19"
              r="1"
            />
          </svg>
        </span>
      </button>
      <ul
        className="dropdown"
      >
        <li
          className="menuItem"
        >
          <button
            onMouseDown={[MockFunction]}
          >
            <span
              className="root"
            >
              <svg
                className="icon"
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 3a2.828 2.828 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5L17 3z"
                />
              </svg>
            </span>
            <span
              className="text"
            >
              Edit item
            </span>
          </button>
        </li>
        <li
          className="menuItem"
        >
          <button
            onMouseDown={[MockFunction]}
          >
            <span
              className="root"
            >
              <svg
                className="icon"
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="3 6 5 6 21 6"
                />
                <path
                  d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                />
              </svg>
            </span>
            <span
              className="text"
            >
              Remove from cart
            </span>
          </button>
        </li>
        <li>
          <AddToListButton
            atwProp1="value1"
            classes={
              Object {
                "root": "addToListButton",
                "root_selected": "addToListButton_selected",
              }
            }
            icon={
              <Icon
                size={16}
                src={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "propTypes": Object {
                      "color": [Function],
                      "size": [Function],
                    },
                    "render": [Function],
                  }
                }
              />
            }
          />
        </li>
      </ul>
    </div>
  </div>
</li>
`;

exports[`renders out of stock product 1`] = `
<li
  className="root"
>
  <span
    className="errorText"
  />
  <div
    className="item"
  >
    <div
      className="imageContainer"
      to="/popular-product"
    >
      <Image
        alt=""
        classes={
          Object {
            "image": "image",
            "root": "imageRoot",
          }
        }
        resource={Object {}}
        width={100}
      />
    </div>
    <div
      className="details"
    >
      <div
        className="name"
      >
        <div
          to="/popular-product"
        >
          
        </div>
      </div>
      <span
        className="price"
      >
        <span>
          $
        </span>
        <span>
          55
        </span>
        <span>
          .
        </span>
        <span>
          00
        </span>
        <mock-FormattedMessage
          defaultMessage=" ea."
          id="product.price"
        />
      </span>
      <span
        className="stockStatusMessage"
      >
        Out-of-stock
      </span>
      <div
        className="quantity"
      >
        <form
          onKeyDown={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="root"
          >
            <label
              className="label"
              htmlFor="123"
            >
              Quantity
            </label>
            <button
              aria-label="Decrease Quantity"
              className="button_decrement"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="root"
              >
                <svg
                  className="icon"
                  fill="none"
                  height={22}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={22}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
            <span
              className="root"
              style={
                Object {
                  "--iconsAfter": 0,
                  "--iconsBefore": 0,
                }
              }
            >
              <span
                className="input"
              >
                <input
                  aria-label="Item Quantity"
                  className="input"
                  id="123"
                  inputMode="numeric"
                  min={0}
                  name="quantity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  pattern="[0-9]*"
                  value={2}
                />
              </span>
              <span
                className="before"
              />
              <span
                className="after"
              />
            </span>
            <p
              className="root"
            />
            <button
              aria-label="Increase Quantity"
              className="button_increment"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="root"
              >
                <svg
                  className="icon"
                  fill="none"
                  height={20}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="12"
                    x2="12"
                    y1="5"
                    y2="19"
                  />
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="root"
    >
      <button
        className="kebab"
        onClick={[Function]}
      >
        <span
          className="root"
        >
          <svg
            className="icon"
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="12"
              cy="5"
              r="1"
            />
            <circle
              cx="12"
              cy="19"
              r="1"
            />
          </svg>
        </span>
      </button>
      <ul
        className="dropdown"
      >
        <li
          className="menuItem"
        >
          <button
            onMouseDown={[MockFunction]}
          >
            <span
              className="root"
            >
              <svg
                className="icon"
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="3 6 5 6 21 6"
                />
                <path
                  d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                />
              </svg>
            </span>
            <span
              className="text"
            >
              Remove from cart
            </span>
          </button>
        </li>
        <li>
          <AddToListButton
            atwProp1="value1"
            classes={
              Object {
                "root": "addToListButton",
                "root_selected": "addToListButton_selected",
              }
            }
            icon={
              <Icon
                size={16}
                src={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "propTypes": Object {
                      "color": [Function],
                      "size": [Function],
                    },
                    "render": [Function],
                  }
                }
              />
            }
          />
        </li>
      </ul>
    </div>
  </div>
</li>
`;

exports[`renders simple product correctly 1`] = `
<li
  className="root"
>
  <span
    className="errorText"
  />
  <div
    className="item"
  >
    <div
      className="imageContainer"
      to="/unittest.html"
    >
      <Image
        alt=""
        classes={
          Object {
            "image": "image",
            "root": "imageRoot",
          }
        }
        resource={Object {}}
        width={100}
      />
    </div>
    <div
      className="details"
    >
      <div
        className="name"
      >
        <div
          to="/unittest.html"
        >
          
        </div>
      </div>
      <span
        className="price"
      >
        <span>
          $
        </span>
        <span>
          1
        </span>
        <span>
          .
        </span>
        <span>
          00
        </span>
        <mock-FormattedMessage
          defaultMessage=" ea."
          id="product.price"
        />
      </span>
      <span
        className="stockStatusMessage"
      >
        
      </span>
      <div
        className="quantity"
      >
        <form
          onKeyDown={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="root"
          >
            <label
              className="label"
              htmlFor="123"
            >
              Quantity
            </label>
            <button
              aria-label="Decrease Quantity"
              className="button_decrement"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              <span
                className="root"
              >
                <svg
                  className="icon"
                  fill="none"
                  height={22}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={22}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
            <span
              className="root"
              style={
                Object {
                  "--iconsAfter": 0,
                  "--iconsBefore": 0,
                }
              }
            >
              <span
                className="input"
              >
                <input
                  aria-label="Item Quantity"
                  className="input"
                  id="123"
                  inputMode="numeric"
                  min={0}
                  name="quantity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  pattern="[0-9]*"
                  value={1}
                />
              </span>
              <span
                className="before"
              />
              <span
                className="after"
              />
            </span>
            <p
              className="root"
            />
            <button
              aria-label="Increase Quantity"
              className="button_increment"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="root"
              >
                <svg
                  className="icon"
                  fill="none"
                  height={20}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="12"
                    x2="12"
                    y1="5"
                    y2="19"
                  />
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="root"
    >
      <button
        className="kebab"
        onClick={[Function]}
      >
        <span
          className="root"
        >
          <svg
            className="icon"
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="12"
              cy="5"
              r="1"
            />
            <circle
              cx="12"
              cy="19"
              r="1"
            />
          </svg>
        </span>
      </button>
      <ul
        className="dropdown"
      >
        <li
          className="menuItem"
        >
          <button
            onMouseDown={[MockFunction]}
          >
            <span
              className="root"
            >
              <svg
                className="icon"
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="3 6 5 6 21 6"
                />
                <path
                  d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                />
              </svg>
            </span>
            <span
              className="text"
            >
              Remove from cart
            </span>
          </button>
        </li>
        <li>
          <AddToListButton
            atwProp1="value1"
            classes={
              Object {
                "root": "addToListButton",
                "root_selected": "addToListButton_selected",
              }
            }
            icon={
              <Icon
                size={16}
                src={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "propTypes": Object {
                      "color": [Function],
                      "size": [Function],
                    },
                    "render": [Function],
                  }
                }
              />
            }
          />
        </li>
      </ul>
    </div>
  </div>
</li>
`;
